service/worker/workerdeployment/workflow.go

```diff
@@ -105,6 +102,25 @@ func (d *WorkflowRunner) listenToSignals(ctx workflow.Context) {
 	d.signalsCompleted = true
 }
 
+func (d *WorkflowRunner) updateVersionSummary(summary *deploymentspb.WorkerDeploymentVersionSummary) {
```

Comment:
```suggestion
func (d *WorkflowRunner) updateVersionSummary(newSummary *deploymentspb.WorkerDeploymentVersionSummary) {
```

nit: I feel like calling this parameter something with "new" or "source" in it makes it clear that the input is not what is being updated. Just reading the function signature as is makes me feel like the summary that is passed in is what will be updated.

Reply:
I think with the latest changes, this concern of yours might have been mitigated - lmk if you still feel the same way and I can change things around 😃